framework/src/eventloop/eventloop_loop.c

```diff
@@ -18,21 +18,82 @@
 /****************************************************************************
  * Included Files
  ****************************************************************************/
+#include <tinyara/config.h>
+
 #include <debug.h>
+#include <unistd.h>
 #include <libtuv/uv.h>
 #include <eventloop/eventloop.h>
 
+#include "eventloop_internal.h"
+
+el_loop_t *g_loop_list[CONFIG_MAX_TASKS];
+
+#define MAX_PID_MASK   (CONFIG_MAX_TASKS - 1)
+#define PID_HASH(pid)  ((pid) & MAX_PID_MASK)
+
+el_loop_t *get_app_loop(void)
+{
+	int index;
+
+	/* Get a loop assigned to own task */
+	index = PID_HASH(getpid());
+	if (index < 0 || index >= CONFIG_MAX_TASKS) {
+		return NULL;
+	}
+
+	if (g_loop_list[index] == NULL) {
+		g_loop_list[index] = (el_loop_t *)EL_ALLOC(sizeof(el_loop_t));
+		if (g_loop_list[index] != NULL) {
+			uv_loop_init(g_loop_list[index]);
+		}
+	}
+
+	return g_loop_list[index];
+}
+
+static void release_app_loop(void)
+{
+	int index;
+
+	index = PID_HASH(getpid());
+	if (index < 0 || index >= CONFIG_MAX_TASKS) {
+		return;
+	}
+
+	if (g_loop_list[index] != NULL) {
+		EL_FREE(g_loop_list[index]);
+	}
+
+	g_loop_list[index] = NULL;
+}
+
 void eventloop_loop_run(void)
 {
-	uv_run(uv_default_loop(), UV_RUN_DEFAULT);
+	el_loop_t *loop;
+
+	loop = get_app_loop();
+	if (loop == NULL) {
+		return;
+	}
+
+	uv_run(loop, UV_RUN_DEFAULT);
```

Comment:
Why don't you add a return value to check whether the loop is successfully started?